### PR TITLE
fix Update settings

### PR DESCRIPTION
**Pull request by megaket4up**
https://github.com/the-control-group/voyager/issues/1669

### CODE REVIEW

File: src/Http/Controllers/VoyagerSettingsController.php [hunk 1]
```diff
@@ -82,9 +82,9 @@ public function update(Request $request)
         $settings = Voyager::model('Setting')->all();
 
         foreach ($settings as $setting) {
-            $content = $this->getContentBasedOnType($request, 'settings', (object) [
+            $content = $this->getContentBasedOnType($request, 'public/settings', (object) [
```
Comment (marktopper) [inline src/Http/Controllers/VoyagerSettingsController.php]: Why the `public/` path if I may ask?

Comment (megaket4up) [reply src/Http/Controllers/VoyagerSettingsController.php]: To upload into `storage/app/public/settings`.

Comment (marktopper) [reply src/Http/Controllers/VoyagerSettingsController.php]: Thanks, will test this today 👍

Comment (megaket4up) [reply src/Http/Controllers/VoyagerSettingsController.php]: Ok

Comment (fletch3555) [reply src/Http/Controllers/VoyagerSettingsController.php]: I think we need a bit more cleanup if this is going to work this way.  That `public` path gets passed in as `$slug`, which is no longer accurate if we're passing a full path.  `$slug` is only used as a prefix to the storage path, so it should still work correctly.  I'd like to know why it broke, though... why this fix is needed in the first place.

Assuming this fix is actually necessary, i propose undoing this so it remains `$slug` (for future use), then adding the `public` prefix to the file and image path in `Controller::getContentBasedOnType()`

File: src/Http/Controllers/Controller.php [hunk 2]
```diff
@@ -167,7 +167,7 @@ public function getContentBasedOnType(Request $request, $slug, $row)
                     $filesPath = [];
                     foreach ($files as $key => $file) {
                         $filename = Str::random(20);
-                        $path = $slug.'/'.date('F').date('Y').'/';
+                        $path = $slug.'/public/'.date('F').date('Y').'/';
```
Comment (fletch3555) [inline src/Http/Controllers/Controller.php]: Wouldn't you want `public/` before `$slug`, not after?

Comment (megaket4up) [reply src/Http/Controllers/Controller.php]: oOps. I'm sorry. Fixed

Comment (fletch3555) [reply src/Http/Controllers/Controller.php]: Thanks!

**Comment by tnylea** [comment]
Just checked this branch and made sure settings are working. Thanks @megaket4up, I did remove the public reference, you will see it should be working now.

I've also fixed the dashboard analytics with this commit 👍 

I'll merge in tomorrow or anyone can merge in after tests have passed.

Thanks :)

**Pull request merged**